functions/src/service/discord.ts

```diff
@@ -1,109 +1,100 @@
   ): Promise<boolean> {
-    return this.sendMessage(settings.discord.channels.alerts, message, files, 'error');
+    return this.sendMessage(settings.discord.channels.alerts, message, 'error', files);
   }
 
-  public static async sendMessageToMaintainers(
+  public async sendMessageToMaintainers(
     message: MessageContent,
     files: FileContent | FileContent[] | undefined = undefined,
   ): Promise<boolean> {
-    return this.sendMessage(settings.discord.channels.maintainers, message, files, 'info');
+    return this.sendMessage(settings.discord.channels.maintainers, message, 'info', files);
   }
 
-  private static async sendMessage(
+  private async sendMessage(
     channelId: string | null,
     messageContent: MessageContent,
-    files: FileContent | FileContent[] | undefined = undefined,
     level: 'debug' | 'info' | 'warn' | 'error' | 'critical',
+    files: FileContent | FileContent[] | undefined = undefined,
   ): Promise<boolean> {
     let isSent = false;
     try {
-      const discord = await this.getInstance();
-
       if (typeof messageContent === 'string') {
         for (const message of Discord.splitMessage(messageContent)) {
-          if (channelId) await discord.createMessage(channelId, message, files);
+          if (channelId) {
+            await this.instance?.createMessage(channelId, message, files);
+          }
 
           // Also send to debug channel
-          await discord.createMessage(
+          await this.instance?.createMessage(
             settings.discord.channels.debug,
             `[${level}] ${message}`,
             files,
           );
         }
       } else {
-        if (channelId) await discord.createMessage(channelId, messageContent, files);
+        if (channelId) {
+          await this.instance?.createMessage(channelId, messageContent, files);
+        }
 
         // Also send to debug channel
         messageContent.content = `[${level}] ${messageContent.content}`;
-        await discord.createMessage(settings.discord.channels.debug, messageContent, files);
+        await this.instance?.createMessage(settings.discord.channels.debug, messageContent, files);
       }
 
       isSent = true;
     } catch (err) {
-      firebase.logger.error('An error occurred while trying to send a message to discord.', err);
+      logger.error('An error occurred while trying to send a message to discord.', err);
     }
 
     return isSent;
   }
 
-  static async getInstance(): Promise<ErisClient> {
-    if (instance) {
-      return instance;
-    }
-
-    instance = new ErisClient(token);
-
-    instance.on('messageCreate', async (message: Message) => {
-      if (message.content === '!ping') {
-        firebase.logger.info('[discord] pong!');
-        await message.channel.createMessage('Pong!');
-      }
-    });
-
-    await instance.connect();
-    await this.becomeReady();
-
-    return instance;
-  }
-
-  static async becomeReady(): Promise<void> {
+  async becomeReady(): Promise<void> {
```

Comment:
This is usually called `init` or `create` or `getInstance`, or made private and called from the constructor.